packages/ember-data/tests/unit/store/push-test.js

```diff
@@ -178,15 +178,18 @@ test("Calling push with a normalized hash containing related records returns a r
       number: '5552121',
       person: 'wat'
     });
-
-    person = store.push('person', {
-      id: 'wat',
-      firstName: 'John',
-      lastName: 'Smith',
-      phoneNumbers: [number1, number2]
-    });
   });
 
+  run(function() {
+    expectDeprecation(function() {
+      person = store.push('person', {
+        id: 'wat',
+        firstName: 'John',
+        lastName: 'Smith',
+        phoneNumbers: [number1, number2]
+      });
+    }, /You tried to push a record 'Person'' with id '<PhoneNumber:.*>' and passed a DS.Model instance as a value for the relationship '1'. You should instead pass a numerical or string id to represent the record./);
```

Comment:
This deprecation warning is updated in #3325 and needs to be updated once the other PR is merged.